addon/components/paper-dialog.js

```diff
@@ -42,20 +41,21 @@ export default Component.extend({
       return '#ember-testing';
     }
     let parent = this.get('defaultedParent');
-    let $parent = $(parent);
-    // If the parent isn't found, assume that it is an id, but that the DOM doesn't
+    let parentEle = typeof parent === "string" ? document.querySelector(parent) : parent;
+    // If the parentEle isn't found, assume that it is an id, but that the DOM doesn't
     // exist yet. This only happens during integration tests or if entire application
     // route is a dialog.
-    if ($parent.length === 0 && parent.charAt(0) === '#') {
+    if (!parentEle && parent.charAt(0) === '#') {
```

Comment:
This creates an impossible condition.
i.e if `!parentEle` then `parent.charAt(0) === '#'` can never be true as well.

I that what we want is:

```js
if (typeof parent === 'string' && parent.charAt(0) === '#') {
```